presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -138,7 +143,7 @@ public ExchangeClientStatus getStatus()
         }
     }
 
-    public synchronized void addLocation(URI location)
+    public synchronized void addLocation(URI location, TaskId sourceTaskId)
```

Comment:
Not sure if we want to flip these two parameters so that it looks more like (key, value).

Reply:
Indeed. I personally prefer not to put it after these confg/gc monitor. But I didn't remember why I put it at second place.. anyway, drag it down :)